### PR TITLE
jwt_verification

### PR DESCRIPTION
**Pull request by codal-umistri**

### CODE REVIEW

File: dbconfig/dbconfig.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+var mysql = require("mysql");
+var connection = mysql.createConnection({
+  host: "localhost",
+  user: process.env.USER,
+  password: process.env.PASSWORD,
+  database: "User",
```
Comment (het-codal) [inline dbconfig/dbconfig.js:6]: Take all DB from env

File: sql/querys.js [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+};
+
+exports.getNote = (userid, table, callback) => {
+  try {
+    connection.query(
+      `SELECT * FROM ${table} WHERE userid = '${userid}'`,
+      (error, results, fields) => {
+        if (error) return callback(error, null);
+
+        console.log(results);
```
Comment (het-codal) [inline sql/querys.js:40]: remove console logs or comment it

File: validations/Validator.js [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+  }
+
+  if (password !== confirm_password) {
+    return res.status(400).json({ message: "Passwords do not match" });
+  }
+
+  next();
+};
+
+exports.validatelogininputs = (req, res, next) => {
```
Comment (het-codal) [inline validations/Validator.js:30]: make this function name as validateLoginInputs

File: validations/Validator.js [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+  }
+
+  if (!validator.isEmail(email)) {
+    return res.status(400).json({ message: "Please enter a valid email" });
+  }
+
+  next();
+};
+
+exports.validateemail = (req, res, next) => {
```
Comment (het-codal) [inline validations/Validator.js:49]: Make function name validateEmail

File: validations/Validator.js [hunk 5]
```diff
@@ -0,0 +1,61 @@
...
+  if (
+    !email ||
+    !password ||
+    email === "undefined" ||
+    password === "undefined"
+  ) {
+    return res.status(400).json({ message: "Please fill in all fields" });
+  }
+
+  if (!validator.isEmail(email)) {
```
Comment (het-codal) [inline validations/Validator.js:42]: make use of common function for validating email

File: validations/Validator.js [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    !name ||
+    !email ||
+    !password ||
+    !confirm_password ||
+    name === "undefined" ||
+    email === "undefined" ||
+    password === "undefined" ||
+    confirm_password === "undefined"
+  ) {
+    return res.status(400).json({ message: "Please fill in all fields" });
```
Comment (het-codal) [inline validations/Validator.js:16]: make sure only one return statement should be their take variable and set message and on base of that return error

File: .gitignore [hunk 7]
```diff
@@ -0,0 +1,9 @@
+# Ignore environment files
+.env
+
+# Ignore the node_modules directory
+node_modules/
+
+# Ignore specific route files in the Routes folder
+Routes/noteRoute.js
```
Comment (het-codal) [inline .gitignore:8]: if you are not using routes file make sure you remove them instade of ignore also create .env.example for .env file only variable should be their

File: Routes/Route.js [hunk 8]
```diff
@@ -0,0 +1,32 @@
...
+  addnotes,
+  deletenotes,
+  getallnotes,
+  updatenotes,
+} = require("../controller/NoteController");
+const auth = require("../middleware/verfifytoken");
+
+const router = express.Router();
+
+router.post("/registration", validateInput, Registration);
```
Comment (het-codal) [inline Routes/Route.js:23]: make sure all method routs (GET, POST, PUT, DELETE)  should be in same manner

**Review by het-codal — CHANGES_REQUESTED** [review]

**Comment by het-codal** [comment]
Also Update readme file and steps for setup, description tech used

File: validations/Validator.js (hunk 6 above)
Comment (codal-umistri) [reply validations/Validator.js:16]: is this correct way :- 

  confirm_password === "undefined"
) {
  errorMessage = "Please fill in all fields";
}

if (errorMessage) {
  return res.status(400).json({ message: errorMessage });
}

**Review by het-codal — APPROVED** [review]

File: controller/NoteController.js [hunk 9]
```diff
@@ -0,0 +1,110 @@
...
+        .json({ message: "Data fetched succesfully", data: result });
+    });
+  } catch (error) {
+    return res.status(500).json({ message: error.message });
+  }
+};
+
+exports.getallnotes = async (req, res) => {
+  try {
+    await getAllNote("NOTES", (error, result) => {
```
Comment (codal-apatel) [inline controller/NoteController.js:43]: Remove await if not needed as you are using callbacks.

Comment (codal-apatel) [reply controller/NoteController.js:43]: Okay. Change callbacks to promise (async, await).

**Review by codal-apatel — APPROVED** [review]

**Pull request merged**